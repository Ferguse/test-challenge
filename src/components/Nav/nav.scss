@import "../../styles/variables.scss";

.nav {
  &-list {
    list-style: none;
    padding: 0;
  }

  &-item {
    margin-right: 20px;
  }

  a {
    text-transform: uppercase;
  }

  &-icon {
    @media screen and (max-width: 600px) {
      display: none;
    }
  }
}
