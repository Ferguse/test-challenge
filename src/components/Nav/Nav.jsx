import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { useCallback, useMemo } from 'react';

import { CATEGORIES } from 'constants/Categories';
import { setCurrentCategory } from 'actions';

import './nav.scss';

export const Nav = () => {
  const dispatch = useDispatch();

  const handleClick = useCallback((name) => () => {
    dispatch(setCurrentCategory(name));
  }, [name]);

  const navList = useMemo(() => (
    CATEGORIES.map(({ name, icon }) => (
      <li key={name} className="nav-item">
        <Link to={`/category/${name}`} onClick={handleClick(name)}>
          <i className={`fi fi-rs-${icon} nav-icon`}></i>
          <span> {name}</span>
        </Link>
      </li>
    ))), []);

  return (
    <nav className="nav">
      <div className="container">
        <ul className="nav-list flex">
          <li className="nav-item">
            <Link to="/">
              <i className="fi fi-rs-home nav-icon"></i>
              <span> Home</span>
            </Link>
          </li>
          {navList}
        </ul>
      </div>
    </nav>
  );
};
