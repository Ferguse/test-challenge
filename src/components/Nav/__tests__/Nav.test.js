import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockedStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { Nav } from '../Nav';

describe('components [Nav]', () => {
  const state = {
    wishList: {},
    categories: {
      entities: [],
    },
    current: {},
  };

  const store = configureMockedStore([thunk])(state);
  const history = createMemoryHistory({ initialEntries: ['/home'] });

  it('Renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history} location={history.location} navigator={history}>
            <Nav />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
