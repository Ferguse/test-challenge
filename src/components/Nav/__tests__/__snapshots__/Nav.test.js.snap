// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components [Nav] Renders 1`] = `
<nav
  className="nav"
>
  <div
    className="container"
  >
    <ul
      className="nav-list flex"
    >
      <li
        className="nav-item"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <i
            className="fi fi-rs-home nav-icon"
          />
          <span>
             Home
          </span>
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/category/bike"
          onClick={[Function]}
        >
          <i
            className="fi fi-rs-bike nav-icon"
          />
          <span>
             
            bike
          </span>
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/category/hiking"
          onClick={[Function]}
        >
          <i
            className="fi fi-rs-hiking nav-icon"
          />
          <span>
             
            hiking
          </span>
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/category/skating"
          onClick={[Function]}
        >
          <i
            className="fi fi-rs-skating nav-icon"
          />
          <span>
             
            skating
          </span>
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
